lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -477,8 +477,8 @@ def create_test_target_copy_resources_script(test_type)
             pod_targets = target.all_dependent_targets
             resource_paths_by_config = target.user_build_configurations.keys.each_with_object({}) do |config, resources_by_config|
               resources_by_config[config] = pod_targets.flat_map do |pod_target|
-                include_test_spec_paths = pod_target == target
-                pod_target.resource_paths(include_test_spec_paths)
+                spec_paths_to_include = pod_target == target ? pod_target.specs.map(&:name) : pod_target.non_test_specs.map(&:name)
+                pod_target.resource_paths.values_at(*spec_paths_to_include).flatten
```

Comment:
should this be `uniq`d ?

Reply:
not for resource paths. Resource bundles are actually separated in installation compared to the actual pod target (and framework paths) where all the subspecs are merged into a single target and hence require `uniq`

Reply:
I mean we could be defensive.